### DISCUSSION TITLE
Why not use the type safe directives syntax, or a global registry like Vue

### DISCUSSION THREAD

**Discussion by yuyi919** [solidjs/solid#658] [Q&A]
#569

**Comment by ryansolid** [comment]
Can you elaborate? I thought our approach to typing directives was Typesafe. I don't use it in the tutorials because we don't use TypeScript in the tutorials, but I'm unclear what the suggestion is.

Can you point me to how Vue does this? One hard requirement is I don't want to touch the language server protocol. Do you call a function that registers them? I'm gathering Vue also does the lookup automatically. One benefit of the current approach is it is done at compile time.

**Comment by yuyi919** [comment]
Sorry, my English is bad.
I think that when use the directives syntax, compiler should automatically import a helper, just like ```delegateEvents```. Related issues #569;
Or we can use a global register function, like ```Vue.use ("model", model)```, and use the string key instead of import target functions in component file.
